tiled/_tests/adapters/test_sql_types.py

```diff
@@ -26,363 +27,234 @@ async def postgres_uri() -> AsyncGenerator[str, None]:
-        ),
-        (
-            pa.schema([("some_int16", pa.int16())]),
-            pa.schema([("some_int16", "int16")]),
-            ["SMALLINT"],
-        ),
-        (
-            pa.schema([("some_uint16", pa.uint16())]),
-            pa.schema([("some_uint16", "int16")]),
-            ["SMALLINT"],
-        ),
-        (
-            pa.schema([("some_int32", pa.int32())]),
-            pa.schema([("some_int32", "int32")]),
-            ["INTEGER"],
-        ),
-        (
-            pa.schema([("some_uint32", pa.uint32())]),
-            pa.schema([("some_uint32", "int32")]),
-            ["INTEGER"],
-        ),
-        (
-            pa.schema([("some_int64", pa.int64())]),
-            pa.schema([("some_int64", "int64")]),
-            ["BIGINT"],
-        ),
-        (
-            pa.schema([("some_uint64", pa.uint64())]),
-            pa.schema([("some_uint64", "int64")]),
-            ["BIGINT"],
-        ),
-        (
-            pa.schema([("some_float16", pa.float16())]),
-            pa.schema([("some_float16", "float")]),
-            ["REAL"],
-        ),
-        (
-            pa.schema([("some_float32", pa.float32())]),
-            pa.schema([("some_float32", "float")]),
-            ["REAL"],
-        ),
-        (
-            pa.schema([("some_float64", pa.float64())]),
-            pa.schema([("some_float64", "double")]),
-            ["DOUBLE PRECISION"],
-        ),
-        (
-            pa.schema([("some_string", pa.string())]),
-            pa.schema([("some_string", "string")]),
-            ["TEXT"],
-        ),
-        (
-            pa.schema([("some_large_string", pa.large_string())]),
-            pa.schema([("some_large_string", "string")]),
-            ["TEXT"],
-        ),
-        (
-            pa.schema([("some_integer_array", pa.list_(pa.int32()))]),
-            pa.schema([("some_integer_array", pa.list_(pa.int32()))]),
-            ["INTEGER ARRAY"],
-        ),
-        (
-            pa.schema([("some_large_integer_array", pa.list_(pa.int64()))]),
-            pa.schema([("some_large_integer_array", pa.list_(pa.int64()))]),
-            ["BIGINT ARRAY"],
-        ),
-        (
-            pa.schema([("some_fixed_size_integer_array", pa.list_(pa.int64(), 2))]),
-            pa.schema([("some_fixed_size_integer_array", pa.list_(pa.int64()))]),
-            ["BIGINT ARRAY"],
-        ),
-    ],
-)
-def test_psql_data_types(
-    actual_schema: pa.Schema,
-    expected_schema: pa.Schema,
-    expected_keywords: List[str],
-    postgres_uri: str,
+@pytest.mark.parametrize("dialect", ["duckdb", "postgresql", "sqlite"])
+@pytest.mark.parametrize("test_case_id", list(TEST_CASES))
+def test_data_types(
+    test_case_id: str,
+    dialect: Literal["postgresql", "sqlite", "duckdb"],
+    request: pytest.FixtureRequest,
 ) -> None:
-    query = arrow_schema_to_create_table(
-        actual_schema, "random_test_table", "postgresql"
-    )
+    table, dialect_results = TEST_CASES[test_case_id]
```

Comment:
I see a string a lot that could create trouble with maintainability. 

```suggestion
    test_table_name = "random_test_table"
    table, dialect_results = TEST_CASES[test_case_id]
```